tensordict/tensordict.py

```diff
@@ -4462,19 +4462,20 @@ def __getitem__(self, item: INDEX_TYPING) -> TensorDictBase:
                 stack_dim=new_stack_dim,
             )
         elif isinstance(item, tuple):
+            # select sub tensordicts
             _sub_item = tuple(
-                _item for i, _item in enumerate(item) if i == self.stack_dim
+                _item for i, _item in enumerate(item) if i != self.stack_dim
             )
-            if len(_sub_item):
-                tensordicts = self.tensordicts[_sub_item[0]]
+
+            if self.stack_dim < len(item):
+                tensordicts = self.tensordicts[item[self.stack_dim]]
                 if isinstance(tensordicts, TensorDictBase):
+                    if _sub_item:
+                        return tensordicts[_sub_item]
```

Comment:
this was missing from the original implementation and was the main reason for the bug afaict